controllers/component_build_controller_pac.go

```diff
@@ -670,6 +673,49 @@ func checkMandatoryFieldsNotEmpty(config map[string][]byte, mandatoryFields []st
 	return nil
 }
 
+func pacRepoAddCustomParams(repository *pacv1alpha1.Repository, namespace *corev1.Namespace) bool {
+	var params []pacv1alpha1.Params
+	if repository.Spec.Params == nil || len(*repository.Spec.Params) == 0 {
+		params = make([]pacv1alpha1.Params, 0)
+	} else {
+		params = *repository.Spec.Params
+	}
+
+	// Avoid adding duplicate params potentially
+	existingParams := make(map[string]byte)
+	existingParamsCount := 0
+	for _, param := range params {
+		// For checking the existence of custom parameters, the value does not matter.
+		existingParams[param.Name] = 0
```

Comment:
If the value does not matter, maybe a `map[string]bool` or `map[string]struct{}` would be cleaner? At first I thought we would be counting things but only up to 255 occurrences

Reply:
Updated.